spring-cloud-gcp-vision/src/main/java/org/springframework/cloud/gcp/vision/CloudVisionTemplate.java

```diff
@@ -97,7 +104,13 @@ public AnnotateImageResponse analyzeImage(Resource imageResource, Feature.Type..
 		BatchAnnotateImagesResponse batchResponse = this.imageAnnotatorClient.batchAnnotateImages(request);
 		List<AnnotateImageResponse> annotateImageResponses = batchResponse.getResponsesList();
 
-		return annotateImageResponses.get(0);
+		if (!annotateImageResponses.isEmpty()) {
+			return annotateImageResponses.get(0);
+		}
+		else {
+			throw new CloudVisionException(
```

Comment:
Should we throw an exception for error responses here too?

Reply:
I think in this case, it's not really an exception. So, let's leave it to the user.